src/test/kotlin/uk/gov/justice/digital/assessments/controller/AssessmentControllerTest.kt

```diff
@@ -10,34 +10,47 @@ import java.time.LocalDateTime
 
 @AutoConfigureWebTestClient
 class AssessmentControllerTest : IntegrationTest() {
-
     private val supervisionId = "SUPERVISION1"
 
     @Test
     fun `access forbidden when no authority`() {
-
         webTestClient.get().uri("/assessments/supervision/$supervisionId")
                 .header("Content-Type", "application/json")
                 .exchange()
                 .expectStatus().isUnauthorized
     }
 
-
     @Test
     fun `create a new assessment returns assessment`() {
+        val assessment = createAssessment("SupervisionId")
 
-        webTestClient.post().uri("/assessments/supervision")
-                .bodyValue(CreateAssessmentDto("SupervisionId"))
+        assertThat(assessment.assessmentId).isEqualTo(1)
+        assertThat(assessment.supervisionId).isEqualTo("SupervisionId")
+        assertThat(assessment.createdDate).isEqualToIgnoringMinutes(LocalDateTime.now())
+    }
+
+    @Test
+    fun `trying to create an assessment when one already exists returns the existing assessment`() {
+        // create assessment
+        val assessment = createAssessment("ExistingSupervisionId")
+
+        // try and create another
+        val existing = createAssessment("ExistingSupervisionId")
+
+        assertThat(existing.assessmentId).isEqualTo(assessment.assessmentId)
+        assertThat(existing.supervisionId).isEqualTo(assessment.supervisionId)
+        assertThat(existing.createdDate).isEqualTo(assessment.createdDate)
+    }
+
+    private fun createAssessment(supervisionId: String): AssessmentDto {
+        val assessment = webTestClient.post().uri("/assessments/supervision")
+                .bodyValue(CreateAssessmentDto(supervisionId))
                 .headers(setAuthorisation())
                 .exchange()
                 .expectStatus().isOk
                 .expectBody<AssessmentDto>()
-                .consumeWith {
-                    val assessment = it.responseBody
-                    assertThat(assessment?.assessmentId).isEqualTo(1)
-                    assertThat(assessment?.supervisionId).isEqualTo("SupervisionId")
-                    assertThat(assessment?.createdDate).isEqualToIgnoringMinutes(LocalDateTime.now())
-                }
+                .returnResult()
+                .responseBody
+        return assessment!!
```

Comment:
We can inline the return statement

Reply:
We can, but it's a longish method and doing an immediate return or a expression method just felt like too much in one go.